modules/math/shared/src/main/scala/gsp/math/skycalc/ImprovedSkyCalcMethods.scala

```diff
@@ -0,0 +1,1413 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gsp.math.skycalc
+
+import java.time.LocalTime
+import java.time.Instant
+import java.time.ZonedDateTime
+import java.time.ZoneId
+
+import scala.annotation.unused
+
+trait ImprovedSkyCalcMethods {
+
+  // defined quantities for apparent place transforms ..
+  protected val XFORM_FROMSTD      = 1
+  protected val XFORM_TOSTDEP: Int = -1
+  protected val XFORM_JUSTPRE      = 1
+  protected val XFORM_DOAPPAR      = 0
+
+  // some (not all) physical, mathematical, and astronomical constants used are defined here.
+  protected val TWOPI     = 6.28318530717959
+  protected val PI_OVER_2 = 1.57079632679490 // From Abramowitz & Stegun
+
+  protected val ARCSEC_IN_RADIAN = 206264.8062471
+  protected val DEG_IN_RADIAN    = 57.2957795130823
+  protected val HRS_IN_RADIAN    = 3.819718634205
+  protected val KMS_AUDAY        = 1731.45683633 // km per sec in 1 AU/day
+
+  protected val SPEED_OF_LIGHT = 299792.458 // in km per sec ... exact.
+
+  protected val J2000 = 2451545.0 // Julian date at standard epoch
+
+  protected val SEC_IN_DAY = 86400.0
+  protected val FLATTEN    = 0.003352813 // flattening of earth, 1/298.257
+
+  protected val EQUAT_RAD = 6378137.0 // equatorial radius of earth, meters
+
+  protected val ASTRO_UNIT = 1.4959787066e11 // 1 AU in meters
+
+  protected val KZEN =
+    0.172 // zenith extinction, mag, for use in lunar sky brightness calculations.
+
+  protected val EARTH_DIFF =
+    0.05 // used in numerical differentiation to find earth velocity
+
+  // Constants needed in etcorr method
+  protected val DELTS: Array[Double] =
+    Array[Double](-2.72, 3.86, 10.46, 17.20, 21.16, 23.62, 24.02, 23.93, 24.33, 26.77, 29.15, 31.07,
+      33.15, 35.73, 40.18, 45.48, 50.54, 54.34, 56.86, 60.78, 62.97)
+
+  private val UT: ZoneId         = ZoneId.of("UT")
+  // private val NanosPerSecond: Double = 1e9
+  private val MillisPerNano: Int =
+    1_000_000 // Deliberately an Int to drop Nano precision
+
+  final protected class DoubleRef(var d: Double) {
+    def this() = {
+      this(0.0)
+    }
+
+    override def toString: String = d.toString
+  }
+
+  protected case class DateTime(
```

Comment:
This is a nitpick but there is another `DateTime` in `java.unit` maybe we could avoid a name clash

Reply:
Hmm.... Let me see if we can get to actually use `java.time.LocalDateTime` and discard this class...

Reply:
I actually replaced its use for `Instant`, which seems like the right thing to do.

Reply:
👍